src/SigninRequest.ts

```diff
@@ -94,6 +93,21 @@ export class SigninRequest {
             client_secret, scope, extraTokenParams,
             skipUserInfo,
         });
+    }
+
+    public async getUrl(): Promise<string> {
+        /* eslint-disable @typescript-eslint/no-unused-vars */
```

Comment:
properly setup local variables, such that we do not need eslint-disable here

Reply:
its a slight pain thanks to the fact we append the `optionalParams` to the query string.

if we want to get rid of the unused vars, they'll become part of `optionalParams`.

which means we'd have to instead have a fixed set of properties we append to the qs and hope the value actually matches the type at runtime. it'd be a change in behaviour since random nonsense people pass in at runtime right now will end up in the qs.